File: LeetCode/Java/Easy/GuessNumberHigherOrLower.java
package Easy;

public class GuessNumberHigherOrLower {

	public void main(String[] args) {

	}

	
	public int guessNumber(int n){
		int low = 1;
		int high = n;
		while (low < high){
			int mid =  low + (high - low)/2;
			if (guess(mid)==0){
				return mid;
			} else if (guess(mid) == 1){
				low = mid + 1;
			} else {
				high = mid;
			}
		}
		return low;
	}

	
	//holder code
	public int guess(int num){
		return num;
	}
}

File: LeetCode/Python/Easy/GuessNumberHigherOrLower.py
class Solution(object):
    def guessNumber(self, n):
        low = 1
        high = n
        while (low < high):
            mid = (high - low) / 2 + low
            result = guess(mid)
            if result == 0:
                return mid
            elif result == -1:
                high = mid - 1
            else:
                low = mid + 1

        return low
